src/tools/analyzer.md

```diff
@@ -0,0 +1,53 @@
+---
+layout: default
+permalink: /tools/analyzer
+title: "Static Analyzer"
+description: "The tool that analyzes and validates your Dart code."
+---
+
+The static analyzer evaluates your Dart code,
+checking for errors and warnings that are specified in the
+[Dart Language Specification](https://www.dartlang.org/docs/spec/).
+
+Many tools in the Dart ecosystem use the [analysis
+server](https://github.com/dart-lang/sdk/tree/master/pkg/analysis_server)
+to perform static analysis. For example:
+
+* Dart and Flutter plugins for IDEs (such as
+  [WebStorm](https://webdev.dartlang.org/tools/webstorm),
+  [IntelliJ](https://www.dartlang.org/tools/jetbrains-plugin), and
+  [Atom](https://atom.io/packages/dartlang))
+* [DartPad](https://www.dartlang.org/tools/dartpad)&mdash;browser-based
+  Dart playground
+* [Dart Dev Compiler (DDC)](https://github.com/dart-lang/sdk/tree/master/pkg/dev_compiler)&mdash;experimental development tool and transpiler for web apps
+* [`dartanalyzer`](https://github.com/dart-lang/sdk/tree/master/pkg/analyzer_cli#dartanalyzer
```

Comment:
The `dartanalyzer` uses `package:analyzer` to perform analysis. The command line tool is worth documenting in its own right.

I think analysis (from the user's perspective) generally happens in two places - from the command-line (via `dartanalyzer`) and from IDEs (powered by the analysis server).

I think it's useful to document how to use the command-line analyzer, which IDEs are available that are based on the analysis server, and how to configure your static analysis (via the analysis options file).

Not 100% sure it's necessary to doc how to use package:analyzer; will make a comment to that effect below.